src/wallet/silentpayments.cpp

```diff
@@ -0,0 +1,179 @@
+#include <wallet/silentpayments.h>
+#include <arith_uint256.h>
+#include <coins.h>
+#include <crypto/common.h>
+#include <crypto/hmac_sha512.h>
+#include <key_io.h>
+#include <undo.h>
+#include <logging.h>
+
+namespace wallet {
+
+Recipient::Recipient(const CKey& scan_seckey, const CKey& spend_seckey)
```

Comment:
This doesn't need to be a struct, could just be functions which the DescScriptPubKeyMan calls as needed

Reply:
https://github.com/achow101/bitcoin/commit/57d72b52ca5a442982907cb09c8b26e1ab5e9d59 can be squashed in here to remove the `Recipient` class.